packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -954,6 +954,41 @@ abstract class CustomClipper<T> {
   String toString() => '$runtimeType';
 }
 
+/// A [CustomClipper] that clips to the outer path of a [ShapeBorder].
+class ShapeBorderClipper extends CustomClipper<Path> {
+  /// Creates a [ShapeBorder] clipper.
+  ///
+  /// The [shapeBorder] argument must not be null.
+  ///
+  /// The [textDirection] argument must be provided non-null if [shapeBorder]
+  /// has a text direction dependency (for example if it is expressed in terms
+  /// of "start" and "end" instead of "left" and "right"). It may be null if
+  /// the border will not need the text direction to paint itself.
+  const ShapeBorderClipper({
+    @required this.shapeBorder,
+    this.textDirection,
+  }) : assert(shapeBorder != null);
+
+  // The shape border whose outer path this clipper clips to.
+  final ShapeBorder shapeBorder;
+
+  /// The text direction to use for getting the outer path for [shapeBorder].
+  ///
+  /// [ShapeBorder]s can depend on the text direction (e.g having a "dent"
+  /// towards the start of the shape).
+  final TextDirection textDirection;
+
+  /// Returns the outer path of [shapeBorder] as the clip.
+  @override
+  Path getClip(Size size)
+    => shapeBorder.getOuterPath(Offset.zero & size, textDirection: textDirection);
```

Comment:
you changed shouldReclip but not getClip :-)

Reply:
done